### DISCUSSION TITLE
How could i get ctx value passed from middlewares in Directives

### DISCUSSION THREAD

**Discussion by wyaow** [99designs/gqlgen#2077] [Q&A] [answered]
### What happened?
1. I used mux as router, and passed some value  in ctx by middleware
``` go
ctx := context.WithValue(r.Context(), “CtxKey”, &CtxVal{
	Cookie:              r.Header.Get("Cookie"),
	CsrfToken: r.Header.Get("CsrfToken"),
})
r = r.WithContext(ctx)
next.ServeHTTP(w, r)
```

2. How could i get this CtxVal of "CtxKey" in c.Directives like below
```
c.Directives.Tag= func(ctx context.Context, obj interface{}, next graphql.Resolver,
 	url string, method *string) (res interface{}, err error) {
        // TODO: here i want get context values from 
	return next(ctx)
}
```

### What did you expect?
get right context value

### Minimal graphql.schema and models to reproduce
--
### versions
 - `go run github.com/99designs/gqlgen version`? 
 - v0.14.0
 - `go version` 
 - 1.14

**Comment by wyaow** [comment, marked as answer]
This can be closed, it's no problem, thank you.

**Comment by droslean** [comment]
@wyaow Can you share the solution?
